examples/virtual-dom-wip/platform/Html/HostJavaScript.roc

```diff
@@ -0,0 +1,279 @@
+     */
+
+    /**
+     * @typedef {{ObjectField: [string, CyclicStructureAccessor]} | {ArrayIndex: [number, CyclicStructureAccessor]} | {SerializableValue: []}} CyclicStructureAccessor
+     */
+
+    /** @type {Array<Node | null>} */
+    const nodes = [];
+
+    /** @type {Array<Listener | null>} */
+    const listeners = [];
+
+    const utf8Decoder = new TextDecoder();
+    const utf8Encoder = new TextEncoder();
+
+    /**
+     * @param {string} wasmFilename
+     */
+    export const init = async (wasmFilename) => {
+      const effects = {
+        /**
+         * @param {number} tagAddr
+         */
+        createElement: (tagAddr) => {
+          const tagName = decodeRocStr(tagAddr);
+          const node = document.createElement(tagName);
+          return insertNode(node);
+        },
+
+        /**
+         * @param {number} contentAddr
+         */
+        createTextNode: (contentAddr) => {
+          const content = decodeRocStr(contentAddr);
+          const node = document.createTextNode(content);
+          return insertNode(node);
+        },
+
+        /**
+         * @param {number} parentId
+         * @param {number} childId
+         */
+        appendChild: (parentId, childId) => {
+          const parent = nodes[parentId];
+          const child = nodes[childId];
+          parent.appendChild(child);
+        },
+
+        /**
+         * @param {number} id
+         */
+        removeNode: (id) => {
+          var _a;
+          const node = nodes[id];
+          nodes[id] = null;
+          node.parentElement.removeChild(node);
+        },
+
+        /**
+         * @param {number} nodeId
+         * @param {number} typeAddr
+         * @param {number} valueAddr
+         */
+        setAttribute: (nodeId, typeAddr, valueAddr) => {
+          const node = nodes[nodeId];
+          const name = decodeRocStr(typeAddr);
+          const value = decodeRocStr(valueAddr);
+          node.setAttribute(name, value);
+        },
+
+        /**
+         * @param {number} nodeId
+         * @param {number} typeAddr
+         */
+        removeAttribute: (nodeId, typeAddr) => {
+          const node = nodes[nodeId];
+          const name = decodeRocStr(typeAddr);
+          node.removeAttribute(name);
+        },
+
+        /**
+         * @param {number} nodeId
+         * @param {number} propNameAddr
+         * @param {number} jsonAddr
+         */
+        setProperty: (nodeId, propNameAddr, jsonAddr) => {
+          const node = nodes[nodeId];
+          const propName = decodeRocStr(propNameAddr);
+          const json = decodeRocListUtf8(jsonAddr);
+          const value = JSON.parse(json);
+          node[propName] = value;
+        },
+
+        /**
+         * @param {number} nodeId
+         * @param {number} propNameAddr
+         */
+        removeProperty: (nodeId, propNameAddr) => {
+          const node = nodes[nodeId];
+          const propName = decodeRocStr(propNameAddr);
+          node[propName] = null;
```

Comment:
Out of curiosity, and please correct me where my understanding is wrong - if `propName` is e.g. `className`, `node.className = null` sets `className="null"`, right? Is it instead preferable to set `className=""`?

Reply:
Damn, you're right! I just tested it in my browser. I was sure that would work. Good catch!
I just checked [Elm's implementation](https://github.com/elm/virtual-dom/blob/1.0.3/src/Elm/Kernel/VirtualDom.js#L910) and it does the equivalent of this:
```js
node[propName] = typeof node[propName] === 'string' ? '' : null;
```